lib/rules/lightness-notation/index.cjs

```diff
@@ -93,28 +102,44 @@ const rule = (primary, _secondaryOptions, context) => {
 
 /**
  * @param {string} value
+ * @param {import('postcss-value-parser').FunctionNode['value']} func
  */
-function asPercentage(value) {
-	const num = parseFloat(value) * 100;
+function asPercentage(value, func) {
+	let num = parseFloat(value);
 
-	if (Number.isInteger(num)) return `${num}%`;
+	if (LIGHTNESS_ZERO_TO_HUNDRED_RANGE_FUNCS.has(func)) {
+		return `${num}%`;
+	}
+
+	if (LIGHTNESS_ZERO_TO_ONE_RANGE_FUNCS.has(func)) {
+		num *= 100;
+	}
+
+	if (Number.isInteger(num)) {
+		return `${num}%`;
+	}
 
 	return `${roundToNumberOfDigits(num, value)}%`;
 }
 
 /**
  * @param {string} value
+ * @param {import('postcss-value-parser').FunctionNode['value']} func
  */
-function asNumber(value) {
+function asNumber(value, func) {
 	const dimension = valueParser.unit(value);
```

Comment:
Why not reuse the parsed number that is returned here?

I think it's better to use that, than to have `parseFloat(value)` and `value.slice(0, -1)`.